engine/service.go

```diff
@@ -179,20 +201,191 @@ func (s *Service) FrontendShutdownCallback(ctx context.Context, req *rpc.EngineR
 	if !exists {
 		return nil, fmt.Errorf("engine %v not found", name)
 	}
-	if err := e.shutdownFrontendCallback(); err != nil {
+	tID, err := e.shutdownFrontendCallback()
+	if err != nil {
 		return nil, errors.Wrapf(err, "failed to execute stopFrontendCallback for %v", name)
 	}
+
+	s.engines[name] = e
+
+	if tID != 0 {
+		if err := s.availableTIDs.ReleaseRange(tID, tID+1); err != nil {
```

Comment:
This should be `ReleaseRange(tID, tID)`. It's all inclusive, see https://github.com/longhorn/longhorn-engine-launcher/blob/processlauncher/util/util.go#L17